crates/admin/src/cluster_controller/service.rs

```diff
@@ -496,6 +541,30 @@ impl<T: TransportConnect> Service<T> {
         Ok(())
     }
 
+    fn seal_and_extend_chain(
+        &self,
+        log_id: LogId,
+        min_version: Version,
+        extension: Option<ChainExtension>,
+        response_tx: oneshot::Sender<anyhow::Result<SealedSegment>>,
+    ) {
+        let task = SealAndExtendTask {
+            log_id,
+            extension,
+            min_version,
+            bifrost: self.bifrost.clone(),
+            metadata_store_client: self.metadata_store_client.clone(),
+            metadata_writer: self.metadata_writer.clone(),
+            observed_cluster_state: self.observed_cluster_state.clone(),
+        };
+
+        _ = TaskCenter::spawn(TaskKind::Disposable, "seal-and-extend", async move {
+            let result = task.run().await;
+            _ = response_tx.send(result);
+            Ok(())
+        });
```

Comment:
Unfortunately I can't because the response_tx has already been moved to the async closure. 
But the good news is that if the response channel was dropped before sending any response, it will be resolved as a shutdown error by the receiving end here https://github.com/restatedev/restate/pull/2375/commits/a90653491806e4d989d9a0d3262a4ddd370345b9#diff-192b3892aacbe2a4e89c2141f20eb6b3e5ebb2050bb567b6a94299b4d7df6a90R282